### ISSUE TITLE
Invalid or unknown policy

### ISSUE THREAD

**Issue by mgoodness** [cloudflare/cfssl#459]
I know GH issues generally aren't meant for tech support, but I'm not sure where else to turn. So please forgive my impertinence.

Trying to configure & use multirootca in a minimal (scratch) Docker container. My [multiroot.conf](https://github.com/cloudflare/cfssl/files/58731/multiroot.conf.txt) and [etcd_config.json](https://github.com/cloudflare/cfssl/files/58735/etcd_config.json.txt) seem to load & validate without issue. The Kubernetes config does too, but I'm not including it here.

Client-side, I'm using the same etcd_config.json and this [etcd-server_csr.json](https://github.com/cloudflare/cfssl/files/58743/etcd-server_csr.json.txt), running:
```
cfssl -loglevel=0 gencert -config=etcd_config.json -profile=peer -label=etcd \
    -hostname="localhost" etcd-server_csr.json
```

Which generates the following debug output:
```
2015/12/10 15:01:52 [DEBUG] loading configuration file from etcd_config.json
2015/12/10 15:01:52 [DEBUG] no default given: using default config
2015/12/10 15:01:52 [DEBUG] match remote in profile to remotes section
2015/12/10 15:01:52 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:01:52 [DEBUG] match remote in profile to remotes section
2015/12/10 15:01:52 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:01:52 [DEBUG] match remote in profile to remotes section
2015/12/10 15:01:52 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:01:52 [DEBUG] validating configuration
2015/12/10 15:01:52 [DEBUG] validate local profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] configuration ok
2015/12/10 15:01:52 [INFO] generate received request
2015/12/10 15:01:52 [INFO] received CSR
2015/12/10 15:01:52 [INFO] generating key: ecdsa-384
2015/12/10 15:01:52 [DEBUG] generate key from request: algo=ecdsa, size=384
2015/12/10 15:01:52 [INFO] encoded CSR
2015/12/10 15:01:52 [DEBUG] validating configuration
2015/12/10 15:01:52 [DEBUG] validate local profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
2015/12/10 15:01:52 [DEBUG] validate remote profile
2015/12/10 15:01:52 [DEBUG] profile is valid
{"code":5200,"message":"Invalid or unknown policy"}
```

If I specify the remote on the command line:
```
cfssl -loglevel=0 gencert -remote=192.168.99.100:8888 -config=etcd_config.json \
    -profile=peer -label=etcd -hostname="localhost" etcd-server_csr.json
```
I get:
```
2015/12/10 15:29:03 [DEBUG] loading configuration file from etcd_config.json
2015/12/10 15:29:03 [DEBUG] no default given: using default config
2015/12/10 15:29:03 [DEBUG] match remote in profile to remotes section
2015/12/10 15:29:03 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:29:03 [DEBUG] match remote in profile to remotes section
2015/12/10 15:29:03 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:29:03 [DEBUG] match remote in profile to remotes section
2015/12/10 15:29:03 [DEBUG] match auth key in profile to auth_keys section
2015/12/10 15:29:03 [DEBUG] validating configuration
2015/12/10 15:29:03 [DEBUG] validate local profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] configuration ok
2015/12/10 15:29:03 [INFO] generate received request
2015/12/10 15:29:03 [INFO] received CSR
2015/12/10 15:29:03 [INFO] generating key: ecdsa-384
2015/12/10 15:29:03 [DEBUG] generate key from request: algo=ecdsa, size=384
2015/12/10 15:29:03 [INFO] encoded CSR
2015/12/10 15:29:03 [DEBUG] validating configuration
2015/12/10 15:29:03 [DEBUG] validate local profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validating configuration
2015/12/10 15:29:03 [DEBUG] validate local profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
2015/12/10 15:29:03 [DEBUG] validate remote profile
2015/12/10 15:29:03 [DEBUG] profile is valid
{"code":7400,"message":"404 page not found\n"}
```
I'm trying to piece this together using whatever information I find around the internet, so it's entirely possible that I'm missing (or misunderstanding) something fundamental. Any guidance would be extremely appreciated.

**Comment by mgoodness** [comment]
Update:

After digging into config/config.go, I discovered that authenticated remotes go into their own section of each signing profile. This is my new, functioning [etcd_config.json](https://github.com/cloudflare/cfssl/files/59777/etcd_config.json.txt). Note that I removed the remotes themselves because including them still resulted in '5200' errors. As far as I can tell, it must be passed on the command line:
```
cfssl -loglevel 0 gencert -config=etcd_config.json -hostname="localhost" -profile=server \
    -label=etcd -remote=192.168.99.100:8888 etcd_csr.json
```
This resolves my issue. Hoping I can circle back at some point and help clean up the docs.
